### PR TITLE
Feat-Go: add StrReplace function

### PR DESCRIPTION
**Pull request by YogiPristiawan**
- Programming langugage: 
- Function(s) that I'm implementing:
- Issue number: 

- [x] I have created unit tests to support my implementation
- [x] I have created in-code documentation (or docstring) to document my code
- [ ] (Optional) I have updated the README.md file to document the function I'm implementing

Ini condisinya banyak sekali, semangat reviewer! :)

### CODE REVIEW

File: go/str_replace.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+var (
+	ErrStrReplaceInvalidParameter = errors.New("must be given parameter correctly")
+)
+
+// The StrReplace() function replaces some characters with some other characters in a string.
+// but it has a bit different than PHP does:
+// 1. it not allowed if parameters has nested slice.
+// 2. parameters can only be string or []string, so if you must convert it first into string like PHP does.
+// 3. you can get a count of the number of replacements in return value.
+
```
Comment (aldy505) [inline go/str_replace.go:18]: newline bikin documentation ajdi putus

File: go/str_replace.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+			switch vFind.Kind() {
+			case reflect.String: // ('replace' == string) && ('str' == string) && ('find' == string)
+				if find.(string) == "" {
+					return strCopy, count, nil
+				}
+
+				if c := strings.Count(strCopy, find.(string)); c > 0 {
+					count += c
+					strCopy = strings.ReplaceAll(strCopy, find.(string), charsReplace)
```
Comment (aldy505) [inline go/str_replace.go:52]: kalo ujung-ujungnya perlu re-assert type nya gini, mending jangan pake reflex, di switch type aja

File: go/str_replace.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+			var strCopy = str.(string)
+
+			switch vFind.Kind() {
+			case reflect.String: // ('replace' == string) && ('str' == string) && ('find' == string)
+				if find.(string) == "" {
+					return strCopy, count, nil
+				}
+
+				if c := strings.Count(strCopy, find.(string)); c > 0 {
```
Comment (aldy505) [inline go/str_replace.go:50]: kalo ujung-ujungnya perlu re-assert type nya gini, mending jangan pake reflex, di switch type aja

File: go/str_replace.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+		// check if given 'str' is a slice
+		switch vStr.Kind() {
+		case reflect.String:
+
+			var strCopy = str.(string)
+
+			switch vFind.Kind() {
+			case reflect.String: // ('replace' == string) && ('str' == string) && ('find' == string)
+				if find.(string) == "" {
```
Comment (aldy505) [inline go/str_replace.go:46]: kalo ujung-ujungnya perlu re-assert type nya gini, mending jangan pake reflex, di switch type aja

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: go/str_replace.go (hunk 4 above)
Comment (YogiPristiawan) [reply go/str_replace.go:46]: oke pak

**Comment by YogiPristiawan** [comment]
sorry salah pencet pak

File: go/str_replace.go (hunk 1 above)
Comment (YogiPristiawan) [reply go/str_replace.go]: done https://github.com/teknologi-umum/pehape/pull/64/commits/5b2c35357dfd4aae2d5420fc69dce82710f2cb6b

File: go/str_replace.go (hunk 2 above)
Comment (YogiPristiawan) [reply go/str_replace.go]: done https://github.com/teknologi-umum/pehape/pull/64/commits/2f1c637223d1b96587640f89fa05691e1b4687ba

File: go/str_replace.go (hunk 3 above)
Comment (YogiPristiawan) [reply go/str_replace.go]: done https://github.com/teknologi-umum/pehape/pull/64/commits/2f1c637223d1b96587640f89fa05691e1b4687ba

File: go/str_replace.go (hunk 4 above)
Comment (YogiPristiawan) [reply go/str_replace.go]: done https://github.com/teknologi-umum/pehape/pull/64/commits/2f1c637223d1b96587640f89fa05691e1b4687ba

**Review by aldy505 — APPROVED** [review]

**Pull request merged**